sdk/metric/labelencoder.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+import (
+	"bytes"
+	"sync"
+
+	"go.opentelemetry.io/otel/api/core"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+)
+
+type defaultLabelEncoder struct {
+	// pool is a pool of labelset builders.
+	pool sync.Pool // *bytes.Buffer
```

Comment:
My first thought was that this is unnecessary. After short thinking my guess is that the use of the pool is to avoid memory allocations when the buffer grows, because we can reuse the unused buffer with an already allocated memory? This could use some short comment.

Reply:
Done. This reduces the number of allocations per LabelSet to 3, but I've commented that it should probably be just 2. The first allocation is the labelset object itself, the second allocation is the buffer.String() here. The third allocation has to do with a call to `sort.Stable()` which I think can probably be fixed.

If I replaced this buffer with a `strings.Builder` we'd have more memory allocations, and if we didn't have the pool we'd have more memory allocations, the benchmark shows all of this.